docs/flow/reference/components/prometheus.remote_write.md

```diff
@@ -0,0 +1,349 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/prometheus.remote_write
+title: prometheus.remote_write
+---
+
+# prometheus.remote_write
+
+`prometheus.remote_write` collects metrics sent from other components into a
+Write-Ahead Log (WAL) and forwards them over the network to a series of
+user-supplied endpoints. Metrics are sent over the network using the
+[Prometheus Remote Write protocol][remote_write-spec].
```

Comment:
We haven't historically, but I thought it might be useful to explain what the protocol is (the spec is really the only formal document which explains it). Happy to take it out if you think it's too much information